### PR TITLE
Rework 'running_managed_controllers' metric interface

### PR DESCRIPTION
**Pull request by jprzychodzen**
#### What this PR does / why we need it:

Allow clear implementation of feature included in #106845.

### Release note

```release-note
NONE
```

#### Special notes for your reviewer:

Metric implemented in #106845 does not have a clear implementation path in KCM/CCM, as approach with `ControllerMetrics` assumes passing this instance into code in multiple places. 

Instead of hiding metric as a variable and moving in a context like approach, let's just follow a pattern of others metric and provide publicly visible functions.

Because `manager` label is an important functionality of this metric, let's initialize it once as selected `manager`.

Instead of using `ControllerMetrics`, using `ControllerManagerMetric` which instances will be owned by CCM and KCM. Those instances will be passed to specific controllers, which will be able to invoke `ControllerStarted` in respective `Run` functions.

Expected implementation pattern - initialize with `NewControllerManagerMetrics(manager string)` in KCM/CCM top level functions (main or similar).

Add `ControllerStarted` and `ControllerStopped` in controller specific `Run` function. Sample location:
 https://github.com/kubernetes/kubernetes/blob/aa0ea624890adace25b6f83dc121d339bff12b69/pkg/controller/cronjob/cronjob_controller.go#L93-L100)

/sig cloudprovider
/kind feature

### CODE REVIEW

File: staging/src/k8s.io/component-base/metrics/prometheus/controllers/metrics.go [hunk 1]
```diff
@@ -21,48 +21,46 @@ import (
...
-		[]string{"name", "manager"},
-	)
-	controllerMetrics := &ControllerMetrics{
-		controllerInstanceCount: controllerInstanceCount,
-	}
+// InitializeControllerMetrics prepares metrics with specific manager.
+func InitializeControllerMetrics(manager string) {
 	once.Do(func() {
-		controllerMetrics.Register()
+		controllerInstanceCount = k8smetrics.NewGaugeVec(
```
Comment (wojtek-t) [inline staging/src/k8s.io/component-base/metrics/prometheus/controllers/metrics.go:36]: We generally:
- define metrics as vars
- do registration in Register() function

take a look at https://github.com/kubernetes/kubernetes/blob/master/staging/src/k8s.io/apiserver/pkg/util/flowcontrol/metrics/metrics.go#L54 as an example

Comment (jprzychodzen) [reply staging/src/k8s.io/component-base/metrics/prometheus/controllers/metrics.go]: Thanks for the pointers and discussion. I've changed code to follow suggested approach.

File: staging/src/k8s.io/component-base/metrics/prometheus/controllers/metrics.go [hunk 2]
```diff
@@ -24,45 +24,45 @@ import (
...
-		controllerInstanceCount: controllerInstanceCount,
+)
+
+// ControllerManagerMetrics is a proxy to set controller manager specific metrics.
+type ControllerManagerMetrics struct {
+	manager string
+}
+
+// NewControllerManagerMetrics create a new ControllerManagerMetrics, with specific manager name.
+func NewControllerManagerMetrics(mgr string) *ControllerManagerMetrics {
```
Comment (wojtek-t) [inline staging/src/k8s.io/component-base/metrics/prometheus/controllers/metrics.go:44]: nit: s/mgr/manager/

[try to avoid shortcuts]

**Review by wojtek-t — COMMENTED** [review]
just one nit - other than that LGTM

**Pull request merged**